src/main.c

```diff
@@ -104,8 +104,14 @@ struct listener_ssl_config_t {
 
 struct listener_config_t {
     int fd;
+#if defined(__linux__) && defined(SO_REUSEPORT)
+    int domain;
+    int so_reuseport;
```

Comment:
Do we need `domain` and `so_reuseport` now that we open new sockets immediately after calling `add_listener`?

Reply:
we can remove these if we remove some assert/if checks。